src/consumer/topic.rs

```diff
@@ -95,45 +96,91 @@ impl<T: DeserializeMessage, Exe: Executor> TopicConsumer<T, Exe> {
                     }
                     break;
                 }
-                Err(ConnectionError::PulsarError(
-                    Some(proto::ServerError::ServiceNotReady),
-                    text,
-                )) => {
-                    if operation_retry_options.max_retries.is_none()
-                        || operation_retry_options.max_retries.unwrap() > current_retries
-                    {
-                        error!("subscribe({}) answered ServiceNotReady, retrying request after {}ms (max_retries = {:?}): {}",
-                        topic, operation_retry_options.retry_delay.as_millis(),
-                        operation_retry_options.max_retries, text.unwrap_or_default());
-
-                        current_retries += 1;
-                        client
-                            .executor
-                            .delay(operation_retry_options.retry_delay)
-                            .await;
-
-                        // we need to look up again the topic's address
-                        let prev = addr;
-                        addr = client.lookup_topic(&topic).await?;
-                        if prev != addr {
-                            info!(
-                                "topic {} moved: previous = {:?}, new = {:?}",
-                                topic, prev, addr
-                            );
+                Err(ConnectionError::PulsarError(Some(err), text)) => {
+                    if matches!(
+                        err,
+                        proto::ServerError::ServiceNotReady | proto::ServerError::ConsumerBusy
+                    ) {
```

Comment:
The `if` condition can be lifted into the match arm.

Reply:
Which one ?

Reply:
It's included in https://github.com/CleverAkanoa/pulsar-rs/pull/2. Resolved.